modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/QueryCheckerExtension.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.util;
+
+import static java.lang.reflect.Modifier.isStatic;
+import static org.apache.ignite.lang.IgniteStringFormatter.format;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
+import org.junit.platform.commons.support.HierarchyTraversalMode;
+
+/**
+ * JUnit extension to inject {@link QueryCheckerFactory} instance into test classes, and ensure the {@link QueryChecker#check()} method is
+ * called for each {@link QueryChecker} instance, which was created via the factory.
+ *
+ * @see InjectQueryCheckerFactory
+ */
+public class QueryCheckerExtension implements BeforeEachCallback, BeforeAllCallback, AfterEachCallback {
+    /** QueryCheckers instances that are managed by this extension. */
+    private static final Set<QueryChecker> QUERY_CHECKERS = new HashSet<>();
+
+    private static final QueryCheckerFactory FACTORY_INSTANCE = new QueryCheckerFactoryImpl(
```

Comment:
@AMashenkov I think we can safely convert all static fields to instances fields w/o losing functionality.

Reply:
I am talking about fields that store state. That applies to methods private static methods as well.

Reply:
Fixed